### PR TITLE
Add config parameter to compile function. Fixes #3028

### LINKED ISSUE #3028: compile should have config parameter
Should it be the first or second (after logger)? What's the precedence?

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by kanitw** [comment]
This is okay, but I still think that making it `{config, logger}` would prevent breaking changes in the future.  (We are creating a breaking change now anyway, better just do it once and for all.)

**Comment by domoritz** [comment]
I prefer it the way it is in this PR and think it will be good for a while.

**Comment by kanitw** [comment]
>  it will be good for a while.

I _strongly_ don't like this part.  If we can make it work forever without hurting much, let's just do it so we don't regret later.  For example, if we want to provide title formatter function, we already have to extend it here.

**Comment by domoritz** [comment]
Okay, I don't feel strong enough about it and changed it.

**Comment by domoritz** [comment]
Done

File: site/usage/compile.md [hunk 1]
```diff
@@ -20,10 +20,16 @@ If you want access to the compiled Vega spec from a Javascript program, you can
 var vgSpec = vl.compile(vlSpec).spec;
 ```
 
-By default, warnings and errors are printed to the JavaScript console. To customize how errors and warnings are handled, you can pass a customize logger to the compile function. 
+### Customized Configuration
```
Comment (kanitw) [inline site/usage/compile.md]: It isn't very clear that compile has  two parameters the second one is `opt` and Opt can have config/logging.  

https://vega.github.io/vega/docs/api/view/#view is one way to do it.

Comment (domoritz) [reply site/usage/compile.md]: You're right. I improved it.

**Pull request merged**